api/services/registration_service.py

```diff
@@ -78,21 +78,24 @@ def update_registration(self, registration, jsonData):
         """Updates the registration data
 
         Args:
-            registration (onject): Takes registration as object
+            registration (object): Takes registration as object
             jsonData (json): Takes request json for updating registration fields
         """
+        if not self.has_default_program_selection:
+            registration.program_id = self.selected_program_id
 
         if registration:
             self.user_id = (
                 self.create_user(jsonData) if self.selected_program_id else None
             )
 
+        if not self.has_default_program_selection:
+            registration.status = models.Registration.RegistrationStatus.COMPLETE
+
```

Comment:
If the user has selected any program then only update the status to complete, else the status should not be updated.